src/viewer/NoViewer.jsx

```diff
@@ -1,26 +1,55 @@
+/* global cozy */
 import React from 'react'
 import classNames from 'classnames'
 import { downloadFile } from 'cozy-client'
 import { isCordova } from 'drive/mobile/lib/device'
-import { openOfflineFile, openOnlineFile } from 'drive/mobile/lib/filesystem'
+import {
+  openOfflineFile,
+  temporarySave,
+  openFileWithCordova
+} from 'drive/mobile/lib/filesystem'
 import { translate } from 'cozy-ui/react/I18n'
 import Button from 'cozy-ui/react/Button'
 
 import styles from './styles'
 
-const OpenWithCordova = ({ t, file }) => (
-  <Button
-    theme="regular"
-    className={styles['pho-viewer-noviewer-download']}
-    onClick={() =>
-      file.isAvailableOffline ? openOfflineFile(file) : openOnlineFile(file)
-    }
-  >
-    {t('Viewer.noviewer.openWith')}
-  </Button>
-)
+class OpenWithCordovaButton extends React.Component {
+  state = {
+    loading: false
+  }
+
+  onClick = () => {
+    const { file, onError } = this.props
+    file.isAvailableOffline
+      ? openOfflineFile(file)
+      : (async () => {
+          this.setState(state => ({ ...state, loading: true }))
+          const response = await cozy.client.files.downloadById(file.id)
```

Comment:
What do you think about moving this code in a new helper in `src/drive/mobile/lib/filesystem.js` ?

Reply:
Same here, i think a function that does `cozy.client.files.downloadById(file.id).then(response => response.blob())` would help in several places.

Reply:
Moving pure functions in a specific file and folder is just a matter of organization.
I admit this is the worst place and I leave it inline because I knew it will catch everybody's attention.
My first idea was to put it in `/src/drive/mobile/lib/filesystem.js` but this file clusters everything about cordova and device's I/O. So I don't know if it is its best place.

What do you thing?

1. move it to `/src/drive/mobile/lib/filesystem.js`?
1. move it to somewhere else?

Reply:
1. for now